runtime/interpreter/encode.go

```diff
@@ -664,25 +749,43 @@ const (
 	//
 	// encodedDictionaryValueLength MUST be updated when new element is added.
 	// It is used to verify encoded dictionaries length during decoding.
-	encodedDictionaryValueLength int = 2
+	encodedDictionaryValueLength = 2
 )
 
 const dictionaryKeyPathPrefix = "k"
 const dictionaryValuePathPrefix = "v"
 
-func (e *Encoder) prepareDictionaryValue(
+// encodeDictionaryValue encodes DictionaryValue as
+// cbor.Tag{
+//			Number: cborTagDictionaryValue,
+//			Content: cborArray{
+//				encodedDictionaryValueKeysFieldKey:    []interface{}(keys),
+//				encodedDictionaryValueEntriesFieldKey: []interface{}(entries),
+//			},
+// }
+func (e *Encoder) encodeDictionaryValue(
 	v *DictionaryValue,
 	path []string,
 	deferrals *EncodingDeferrals,
-) (
-	interface{},
-	error,
-) {
+) error {
+	// Encode CBOR tag head
+	err := e.enc.EncodeTagHead(cborTagDictionaryValue)
+	if err != nil {
+		return err
+	}
+
+	// Encode CBOR array head with 2 elements
```

Comment:
This looks good, only one concern: 

We don't explicitly set specific array indices like before anymore, i.e. we don't use `encodedDictionaryValueKeysFieldKey` and `encodedDictionaryValueEntriesFieldKey` here, but sequentially encode the values.  This could lead to an (accidental) disconnect with the decoder.

Not sure how to prevent this, maybe just add a comment warning that we need to keep the field keys in sync.

Same concern for the other encoding functions, e.g. the composite encoding

Reply:
Yeah, I agree and didn't see a simple way to enforce it.  I'll go through the code and add comments with field key constants so we can match fields to their indices.

Reply:
Perfect, thank you! 🙏  I know this is quite some work, but will hopefully save us some troubles down the road

Reply:
All done with requested comments (sorry I didn't do this before PR).  The comments start with "// Encode " like this:

```Go
	// Encode location at array index encodedCompositeValueLocationFieldKey
	err = e.encodeLocation(v.Location)
```

Reply:
Nice!